### ISSUE TITLE
UPL Accuracy Zero 

### ISSUE THREAD

**Issue by muhammad-shahid0749** [tonyhuang2022/UPL#10]
Good Day, 

I have trained the UPL Model on sscaltech01 dataset, but got zero accuracy at the end.  I want to ask do I need to train the model on all the datasets to get the results?

Moreover, I was planning to Train the model on some other different Datasets as well which are not used in this paper, so in that case should I also train the model with default datasets as well ?

**Comment by muhammad-shahid0749** [comment]
One thing which I noticed is the paths of images are inappropriate. 
In your model you have mentioned ip='./data/"+ip.split('/data/')[0]


but the image paths in my windows are being read as "some\things\data\".

**Comment by tonyhuang2022** [comment]
Sorry for the delay response.
For your first question, you do not need train the UPL on all datasets. Maybe your environments or config is not right.
For second, no , you needn't. You can follow the datasets instruction of CoOp and process your own data to train the UPL. You do not need other datasets at all.

**Comment by tonyhuang2022** [comment]
Yes, if you want to run it on Windows PC, you may need to adjust this part.

**Comment by StifleTechnologies** [comment]
Thank You so much for your reply. I am able to run he code on Google Collab. 
I have some additional questions, 
1.How to print/view the results shown in figure 9 of your paper. (Which function should I run after testing and training)?

2.Can we use GAN as vision encoder/decoder for feature extraction (dataset has minor difference for each class) rather than the CLIP vision encoder in your model?

**Comment by muhammad-shahid0749** [comment]
Thank You so much for your reply. I am able to run he code on Google Collab.
I have some additional questions,
1.How to print/view the results shown in figure 9 of your paper. (Which function should I run after testing and training)?

2.Can we use GAN as vision encoder/decoder for feature extraction (dataset has minor difference for each class) rather than the CLIP vision encoder in your model?

**Comment by tonyhuang2022** [comment]
1. We didn't provide the plot function. If you want to plot this, you need to obtain the probability and plot it by yourself.
2. I don't know. But the feature space of GAN may be different from that of CLIP text encoder.
